ranger/core/runner.py

```diff
@@ -226,13 +226,26 @@ def __call__(
                 wait_for_enter = True
         if 'r' in context.flags:
             # TODO: make 'r' flag work with pipes
-            if 'sudo' not in get_executables():
-                return self._log("Can not run with 'r' flag, sudo is not installed!")
+            # XXX: is there any way to make it better?
+            root_commands = ['sudo', 'doas', 'run0']
+            root_command = None
+            for tested_command in root_commands:
+                if tested_command not in get_executables():
+                    continue
+                else:
+                    root_command = tested_command
+                    break
             f_flag = ('f' in context.flags)
-            if isinstance(action, str):
-                action = 'sudo ' + (f_flag and '-b ' or '') + action
-            else:
-                action = ['sudo'] + (f_flag and ['-b'] or []) + action
+            # TODO: 'f' flag for doas/run0
+            match(root_command):
```

Comment:
Note: `match` statements are added in Python 3.10

AFAIK (based on setup.py content), ranger is supporting Python 3.1 and even 2.7 :disappointed: